CMakeLists.txt

```diff
@@ -64,11 +68,25 @@ endif()
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
+if(MSVC AND IGNORE_PYTHON_DEBUG_LIBRARY)
+  # A typical build on Windows needs to have Release and Debug libraries but most
+  # people don't have the python debug library. Hence, we ignore it here and link
+  # against the Release python library even in debug mode.
+  set(PYTHON3_DEBUG_LIBRARY_NAME python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR}_d.lib)
+  set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${CMAKE_SHARED_LINKER_FLAGS} /NODEFAULTLIB:${PYTHON3_DEBUG_LIBRARY_NAME}")
+  set(CMAKE_MODULE_LINKER_FLAGS_DEBUG "${CMAKE_MODULE_LINKER_FLAGS_DEBUG} /NODEFAULTLIB:${PYTHON3_DEBUG_LIBRARY_NAME}")
+  set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} /NODEFAULTLIB:${PYTHON3_DEBUG_LIBRARY_NAME}")
+endif()
 add_library(${PROJECT_NAME} SHARED ${MODELS_SOURCES} ${OPERATOR_SOURCES} ${IMAGE_SOURCES})
 target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES} ${PNG_LIBRARY} ${JPEG_LIBRARIES} Python3::Python)
 set_target_properties(${PROJECT_NAME} PROPERTIES
   EXPORT_NAME TorchVision
   INSTALL_RPATH ${TORCH_INSTALL_PREFIX}/lib)
+if(MSVC)
+  # This is required for consuming projects otherwise, the linker
+  # will strip the operators registration symbols.
+  target_link_options(${PROJECT_NAME} INTERFACE /OPT:NOREF)
```

Comment:
I hope torchvision will remain small like now forever. When it grows larger, this will definitely stop working.

Reply:
Related to this, we thought that https://github.com/pytorch/vision/pull/2798 had fixed the linker issues once and for all, but we still need it for Windows as can be seen in https://github.com/pytorch/vision/blob/ce342580f3ae3f937fa5389c48b82a827b4804df/test/tracing/frcnn/test_frcnn_tracing.cpp#L7-L11 Do you have an idea on how to properly fix this?

Reply:
I didn't have time to look into it carefully. So at least for now, I will approve this change.

Reply:
cc @bmanga